test/controllers/sections/items/ItemWineGrowingZoneControllerSpec.scala

```diff
@@ -0,0 +1,113 @@
+import forms.sections.items.ItemWineGrowingZoneFormProvider
+import mocks.services.{MockGetCnCodeInformationService, MockUserAnswersService}
+import models.sections.items.ItemWineGrowingZone
+import models.{NormalMode, UserAnswers}
+import navigation.FakeNavigators.FakeItemsNavigator
+import pages.sections.items.{ItemExciseProductCodePage, ItemWineGrowingZonePage}
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Helpers}
+import views.html.sections.items.ItemWineGrowingZoneView
+
+import scala.concurrent.Future
+
+class ItemWineGrowingZoneControllerSpec extends SpecBase with MockUserAnswersService with MockGetCnCodeInformationService {
+
+  lazy val formProvider = new ItemWineGrowingZoneFormProvider()
+  lazy val form = formProvider()
+  lazy val view = app.injector.instanceOf[ItemWineGrowingZoneView]
+
+  lazy val submitUrl = routes.ItemWineGrowingZoneController.onSubmit(testErn, testDraftId, testIndex1, NormalMode)
+
+  lazy val defaultUserAnswers = emptyUserAnswers.set(ItemExciseProductCodePage(testIndex1), "W200")
+
+  class Test(val userAnswers: Option[UserAnswers]) {
+    lazy val request: FakeRequest[AnyContentAsEmpty.type] = FakeRequest()
+
+    lazy val controller = new ItemWineGrowingZoneController(
+      messagesApi = messagesApi,
+      userAnswersService = mockUserAnswersService,
+      navigator = new FakeItemsNavigator(testOnwardRoute),
+      auth = fakeAuthAction,
+      getData = new FakeDataRetrievalAction(userAnswers, Some(testMinTraderKnownFacts)),
+      requireData = dataRequiredAction,
+      userAllowList = fakeUserAllowListAction,
+      formProvider = formProvider,
+      controllerComponents = Helpers.stubMessagesControllerComponents(),
+      view = view,
+      cnCodeInformationService = mockGetCnCodeInformationService
+    )
+  }
+
+  "ItemWineGrowingZone Controller" - {
+
+    "must return OK and the correct view for a GET" in new Test(Some(defaultUserAnswers)) {
+      val result = controller.onPageLoad(testErn, testDraftId, testIndex1, NormalMode)(request)
+
+      status(result) mustEqual OK
+      contentAsString(result) mustEqual view(form, submitUrl)(dataRequest(request, userAnswers.get), messages(request)).toString
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in new Test(Some(
+      defaultUserAnswers.set(ItemWineGrowingZonePage(testIndex1), ItemWineGrowingZone.values.head)
+    )) {
+      val result = controller.onPageLoad(testErn, testDraftId, testIndex1, NormalMode)(request)
+
+      status(result) mustEqual OK
+      contentAsString(result) mustEqual view(form.fill(ItemWineGrowingZone.values.head), submitUrl)(dataRequest(request, userAnswers.get), messages(request)).toString
+    }
+
+    "must redirect to the next page when valid data is submitted" in new Test(Some(defaultUserAnswers)) {
+
+      MockUserAnswersService.set().returns(Future.successful(defaultUserAnswers))
```

Comment:
Is it worth asserting that the user answers have been called properly (e.g. `MockUserAnswersService.set(defaultUserAnswers.set(ItemWineGrowingZonePage(testIndex1), ItemWineGrowingZone.values.head)`)?

Reply:
Yep, good shout 👍